### ISSUE TITLE
MATPLOTLIBDATA Deprecation Warning

### ISSUE THREAD

**Issue by mrclary** [ronaldoussoren/py2app#336]
I get a deprecation warning in apps built with `py2app`
```
The MATPLOTLIBDATA environment variable was deprecated in Matplotlib 3.1 and will be removed in 3.3.
```
This is a result of `recipes/matplotlib_prescript.py` setting the deprecated environment variable
```python
import os

os.environ["MATPLOTLIBDATA"] = os.path.join(os.environ["RESOURCEPATH"], "mpl-data")
```
and the same lines appearing in `Contents/Resources/__boot__.py`.

It seems that the prescript can be conditionally applied depending on the installed version of `matplotlib`

**Comment by ronaldoussoren** [comment]
A quick google seems to indicate that there is no replacement: https://matplotlib.org/3.3.4/api/api_changes.html (look for MATPLOTLIBDATA on that page).  The source seems to indicate the same (https://matplotlib.org/3.3.4/_modules/matplotlib.html#get_data_path).  My initial conclusion: the recipe should be changed to include Matplotlib outside of the site-packages.zip file (e.g. automatically use ``packages=["matplotlib"]``)

**Comment by mrclary** [comment]
@ronaldoussoren, I agree that the recipe should force Matplotlib outside of the site-packages.zip. The environment variable assignment should also be removed, or conditioned on Matplotlib >= 3.3.1.
